src/firewall/functions.py

```diff
@@ -673,6 +673,19 @@ def wrong_args_for_callable(fcn, *a, **kw):
     return True
 
 
+def iter_reiterable(iterable):
```

Comment:
Most callers probably are just lists, so re-iterating them is fine. It would be wasteful to clone the list for that. But more importantly, on the next review, somebody thinks "oh, this clone is unneecessary". Then I would have to add a code comment explaining why the clone is even there. At this point, I wouldn't clone the list.

But that makes the function less general and the caller needs to make sure not to mess up (in non-obvious ways). The function no longer accepts arbitrary iterables, only iterables that can be iterated more than once. So, maybe I should add a code comment there that indicates that this function can only be called with certain iterables. But all callers currently are lists (probably?), so I wouldn't want to add such a code comment either.

`iter_reiterable()` solves this. More than anything, it clearly expresses the intent of making the implementation safe. The caller no longer need to worry. The reviewer of the code also sees clearly why this is done, and can reason why this is the correct thing to do.

It doesn't matter whether all callers today(!!) pass a `list()`. Using `iter_reiterable()` in a function makes the function more accepting to input and less of a foot gun.